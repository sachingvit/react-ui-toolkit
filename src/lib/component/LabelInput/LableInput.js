import React from 'react'
import Label from './../Label/Label'
import NoContent from './../NoContent/NoContent'


function LableInput(props) {

    const {
        id,
        layoutClasses,
        labelClasses,
        // inline = false,
        fieldClasses: classes,
        tyep = 'input',
        show = true,
        label
    } = props

    function onChangeHandler(event) {
        console.log('chnage event', event)
    }

    function onFocusHandler(event) {
        console.log('focus event', event)
    }

    if (!show) {
        return <NoContent />
    }

    return (
        <div className={layoutClasses}>
            <div><Label id={id} label={label} labelClasses={labelClasses} /></div>
            <div data-ui-toolkit-id={id} className={classes}>
                <input id={id} type={tyep} onFocus={onFocusHandler} onChange={onChangeHandler} />
            </div>
        </div>
    )
}

export default LableInput
