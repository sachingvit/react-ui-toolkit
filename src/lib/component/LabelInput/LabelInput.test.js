import LabelInput from "./LableInput"
import React from "react";
import {rea} from "react-dom";
import TestRenderer from "react-test-renderer";
import LableInput from "./LableInput";


describe("component > LabelInput", () => {

    it('render', () => {
        const props = {
            id: 'input-field',
            layoutClasses: 'label-class',
            labelClasses: 'label-class',
            // inline = false,
            fieldClasses: 'field-class',
            show: true,
            label: 'First Name'
        }
        expect(TestRenderer.create(<LableInput {...props} />).toJSON()).toMatchSnapshot()
    })

    it('render as show false', () => {
        const props = {
            id: 'input-field',
            layoutClasses: 'label-class',
            labelClasses: 'label-class',
            // inline = false,
            fieldClasses: 'field-class',
            show: false,
            label: 'First Name'
        }
        expect(TestRenderer.create(<LableInput {...props} />).toJSON()).toMatchSnapshot()
    })
})