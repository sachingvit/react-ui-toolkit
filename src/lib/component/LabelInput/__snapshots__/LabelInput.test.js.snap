// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component > LabelInput render 1`] = `
<div
  className="label-class"
>
  <div />
  <div
    className="field-class"
    data-ui-toolkit-id="input-field"
  >
    <input
      id="input-field"
      onChange={[Function]}
      onFocus={[Function]}
      type="input"
    />
  </div>
</div>
`;

exports[`component > LabelInput render as show false 1`] = `null`;
